src/excercise7.js

```diff
@@ -0,0 +1,162 @@
+/**
+ * Created by ttn on 28/12/16.
+ */
+/**
+ * Create a User Registration form with fields ( FirstName, LastName, EmailId, Contact Number, Password and Confirm password).
+ * Form should have a submit button. Apply form validation to incorporate following rules.
+ *
+ */
+
+
+import React from 'react'
+
+export default class UserRegisration extends React.Component {
+
+  constructor(props){
+    super(props)
+    this.state = {
+      firstNameTxt : "",
+      lastNameTxt : "",
+      emailTxt : "",
+      contactNumberTxt : "",
+      passwdTxt : ""
+    }
+
+    this.submitForm = this.submitForm.bind(this)
+
+  }
+
+
+  validateEmail(value){
+
+    let re = /^(([^<>()[\]\\.,;:\s@\"]+(\.[^<>()[\]\\.,;:\s@\"]+)*)|(\".+\"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
+    return re.test(value);
+  }
+
+
+  submitForm(){
+
+    if(this.refs.firstNameRef.value > 0 && this.refs.lastNameRef.value > 0 ){
+      if(this.refs.firstNameRef.value === this.refs.lastNameRef.value){
+        this.setState({
+          firstNameTxt : "Validation Failed.",
+          lastNameTxt : "Validation Failed."
+        })
+      } else {
+        this.setState({
+          firstNameTxt : "Validation Passed.",
+          lastNameTxt : "Validation Passed."
+        })
+      }
+    } else {
+      this.setState({
+        firstNameTxt : "Validation Failed, Field is mandatory.",
+        lastNameTxt : "Validation Failed, Field is mandatory."
+      })
+    }
+
+
+    if(this.refs.passwdRef.value === this.refs.confPasswdRef.value){
+      this.setState({
+        passwdTxt : "Validation Passed."
+      })
+    } else {
+      this.setState({
+        passwdTxt : "Validation Failed."
+      })
+    }
+
+    let contactNo = this.refs.contactNumberRef.value
+
+    if(contactNo.length === 10){
+      this.setState({
+        contactNumberTxt : "Validation Passed."
+      })
+    } else {
+      this.setState({
+        contactNumberTxt : "Validation Failed."
+      })
+    }
+
+    let emailId = this.refs.emailRef.value
+
+    if (this.validateEmail(emailId)) {
+      this.setState({
+        emailTxt : "Validation Passed."
+      })
+    } else {
+      this.setState({
+        emailTxt : "Validation Failed."
+      })
+    }
+
+  }
+
+  render(){
+    let style = {
+      color : "red"
+    }
+
+    return(
+      <div>
+        <table>
+          <tr>
+            <td>First Name : </td>
+            <td>
+              <input ref="firstNameRef" type="text" name="firstName" />
```

Comment:
could have created a wrapper component for input and passed validation regex to that and encapsulated the logic in one component.
This way, you just needed to add the component and pass validation function and the it will be the components responsibility to display validation message.